### ISSUE TITLE
opam 2.0rc fails to fetch MetaOCaml_309_alpha_030.tar.gz when updating

### ISSUE THREAD

**Issue by XVilka** [ocaml/opam-repository#11392]
```
=-=- Upgrading repositories from older opam format =-=-=-=-=-=-=-=-=-=-=-=-=-=-=
Upgrading repository "local"...
Fetching patches of ocaml-variants.3.07+1 to check their hashes...
Fetching patches of ocaml-variants.3.07+2 to check their hashes...
[ERROR] Curl failed: "/usr/bin/curl --write-out %{http_code}\\n --retry 3 --retry-delay 2 --user-agent opam/2.0.0~rc -L -o /tmp/opam-23585-09b4f7/MetaOCaml_309_alpha_030.tar.gz.part
        http://web.archive.org/web/20120209151915/www.metaocaml.org/dist/old/MetaOCaml_309_alpha_030.tar.gz" exited with code 7
[ERROR] Could not get the archive of ocaml-variants.3.09.1+metaocaml, skipping
```
Is this version really needed?

**Comment by yallop** [comment]
I'd prefer to keep that version around.

Is the network issue still occurring for you?  The download worked for me just now.

**Comment by XVilka** [comment]
This one repeats every single time.

**Comment by yallop** [comment]
There's a fix in #11425.
